presto-hive/src/test/java/com/facebook/presto/hive/TestHiveIntegrationSmokeTest.java

```diff
@@ -2505,6 +2506,40 @@ public void testMaterializedPartitioning()
 
         // join
         assertQuery(materializeExchangesSession, "SELECT * FROM (lineitem JOIN orders ON lineitem.orderkey = orders.orderkey) x", assertRemoteMaterializedExchangesCount(2));
+
+        // join a bucketed table with an unbucketed table
+        assertUpdate(
+                // bucket count has to be different from materialized bucket number
+                "CREATE TABLE test_bucketed_lineitem1\n" +
+                        "WITH (bucket_count = 11, bucketed_by = ARRAY['orderkey']) AS\n" +
+                        "SELECT * FROM lineitem",
+                "SELECT count(*) from lineitem");
+        assertQuery(
+                materializeExchangesSession,
+                "SELECT * FROM test_bucketed_lineitem1 JOIN orders ON test_bucketed_lineitem1.orderkey = orders.orderkey",
+                "SELECT * FROM lineitem JOIN orders ON lineitem.orderkey = orders.orderkey",
+                assertRemoteMaterializedExchangesCount(1));
+
+        // join a bucketed table with an unbucketed table; the join has constant pushdown
+        assertUpdate(
+                // bucket count has to be different from materialized bucket number
+                "CREATE TABLE test_bucketed_lineitem2\n" +
+                        "WITH (bucket_count = 11, bucketed_by = ARRAY['partkey', 'suppkey']) AS\n" +
+                        "SELECT * FROM lineitem",
+                "SELECT count(*) from lineitem");
+        assertQuery(
+                materializeExchangesSession,
+                "SELECT * \n" +
+                        "FROM test_bucketed_lineitem2 JOIN partsupp\n" +
+                        "ON test_bucketed_lineitem2.partkey = partsupp.partkey AND\n" +
+                        "test_bucketed_lineitem2.suppkey = partsupp.suppkey\n" +
+                        "WHERE test_bucketed_lineitem2.suppkey = 42",
```

Comment:
I wonder if we should add more tests that exercise this corner case, as it may introduce a correctness issue that is very hard to discover:

- Constants of different types (VARCHAR / DOUBLE)
- More than a single constant
- Different constant position in a hash
  - [column, constant, column]
  - [constant, constant, column]
  - [column, constant, constant]